### PR TITLE
Refactor PageLoader

### PR DESCRIPTION
**Pull request by revonateB0T**

### CODE REVIEW

**Comment by FooIbar** [comment]
```
FATAL EXCEPTION: main
Process: moe.tarsin.ehviewer.debug, PID: 22787
java.lang.VerifyError: Verifier rejected class com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4$1: java.lang.Object com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4$1.doRealWork(int, kotlin.coroutines.Continuation) failed to verify: java.lang.Object com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4$1.doRealWork(int, kotlin.coroutines.Continuation): [0x19] cannot access instance field java.lang.Object com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4$1$3.result from object of type Precise Reference: com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$1$1$4$1$3 (declaration of 'com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4$1' appears in /data/data/moe.tarsin.ehviewer.debug/code_cache/.overlay/base.apk/classes12.dex)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4.invokeSuspend(ArchivePageLoader.kt:73)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4.invoke(Unknown Source:8)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1$4.invoke(Unknown Source:2)
	at arrow.fx.coroutines.ResourceExtensionsKt$autoCloseable$2.invokeSuspend(ResourceExtensions.kt:144)
	at arrow.fx.coroutines.ResourceExtensionsKt$autoCloseable$2.invoke(Unknown Source:8)
	at arrow.fx.coroutines.ResourceExtensionsKt$autoCloseable$2.invoke(Unknown Source:4)
	at arrow.fx.coroutines.ResourceScopeImpl$install$$inlined$bracketCase$1.invokeSuspend(Bracket.kt:258)
	at arrow.fx.coroutines.ResourceScopeImpl$install$$inlined$bracketCase$1.invoke(Unknown Source:8)
	at arrow.fx.coroutines.ResourceScopeImpl$install$$inlined$bracketCase$1.invoke(Unknown Source:4)
	at kotlinx.coroutines.intrinsics.UndispatchedKt.startUndispatchedOrReturn(Undispatched.kt:42)
	at kotlinx.coroutines.BuildersKt__Builders_commonKt.withContext(Builders.common.kt:164)
	at kotlinx.coroutines.BuildersKt.withContext(Unknown Source:1)
	at arrow.fx.coroutines.ResourceScopeImpl.install(Resource.kt:1227)
	at arrow.fx.coroutines.ResourceExtensionsKt.autoCloseable(ResourceExtensions.kt:144)
	at arrow.fx.coroutines.ResourceExtensionsKt.autoCloseable$default(ResourceExtensions.kt:141)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1.invokeSuspend(ArchivePageLoader.kt:72)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1.invoke(Unknown Source:8)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3$1.invoke(Unknown Source:4)
	at arrow.fx.coroutines.Resource.use(Resource.kt:311)
	at arrow.fx.coroutines.ResourceKt.resourceScope(Resource.kt:896)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3.invokeSuspend(ArchivePageLoader.kt:61)
	at com.hippo.ehviewer.ui.reader.ReaderScreenKt$ReaderScreen$1$pageLoader$2$1$1$1$invokeSuspend$lambda$1$$inlined$usePageLoader-lT7Y-FI$3.invoke(Unknown Source:8)

```

**Comment by FooIbar** [comment]
Repro: opan a gallery then open another from file manager

File: app/src/main/kotlin/com/hippo/ehviewer/gallery/MetaPageLoader.kt [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.update
+
+private const val MAX_CACHE_SIZE = 512 * 1024 * 1024
+private const val MIN_CACHE_SIZE = 128 * 1024 * 1024
+
+abstract class MetaPageLoader : AutoCloseable {
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/gallery/MetaPageLoader.kt:17]: What's the benefit of keeping this instead of merging with `PageLoader`?

Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/gallery/MetaPageLoader.kt:17]: IDK, I will merge if you like it

Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/gallery/MetaPageLoader.kt]: Done

**Review by FooIbar — APPROVED** [review]
It looks much cleaner

**Pull request merged**